bigtable-hbase-1.x-parent/bigtable-hbase-1.x/src/test/java/com/google/cloud/bigtable/hbase1_x/TestMetrics.java

```diff
@@ -0,0 +1,231 @@
+  private static final String FULL_TABLE_NAME =
+      NameUtil.formatTableName(TEST_PROJECT_ID, TEST_INSTANCE_ID, TABLE_NAME.getNameAsString());
+  private static Server server;
+  private static int dataPort;
+  private static final AtomicInteger callCount = new AtomicInteger(1);
+
+  private static final FakeDataService fakeDataService = new FakeDataService();
+  private BigtableConnection connection;
+
+  @Before
+  public void setUp() throws IOException {
+    Configuration configuration = new Configuration(false);
+    configuration.set(BigtableOptionsFactory.PROJECT_ID_KEY, TEST_PROJECT_ID);
+    configuration.set(BigtableOptionsFactory.INSTANCE_ID_KEY, TEST_INSTANCE_ID);
+    configuration.set(BigtableOptionsFactory.BIGTABLE_NULL_CREDENTIAL_ENABLE_KEY, "true");
+    configuration.set(BigtableOptionsFactory.BIGTABLE_DATA_CHANNEL_COUNT_KEY, "1");
+    configuration.set(BigtableOptionsFactory.BIGTABLE_EMULATOR_HOST_KEY, "localhost:" + dataPort);
+    configuration.set(BigtableOptionsFactory.BIGTABLE_USE_GCJ_CLIENT, "true");
+    connection = new BigtableConnection(configuration);
+  }
+
+  @BeforeClass
+  public static void setUpServer() throws IOException {
+    try (ServerSocket s = new ServerSocket(0)) {
+      dataPort = s.getLocalPort();
+    }
+    server = ServerBuilder.forPort(dataPort).addService(fakeDataService).build();
+    server.start();
+  }
+
+  @AfterClass
+  public static void tearDownServer() throws InterruptedException {
+    if (server != null) {
+      server.shutdownNow();
+      server.awaitTermination();
+    }
+  }
+
+  @After
+  public void tearDown() throws IOException {
+    connection.close();
+  }
+
+  @Test
+  public void readRows() throws IOException, InterruptedException {
+    FakeMetricRegistry fakeMetricRegistry = new FakeMetricRegistry();
+
+    BigtableClientMetrics.setMetricRegistry(fakeMetricRegistry);
+    Table table = connection.getTable(TABLE_NAME);
+
+    long readRowsStart = System.currentTimeMillis();
+    Result result = table.get(new Get(new byte[2]));
+    long readRowsTime = System.currentTimeMillis() - readRowsStart;
+
+    ReadRowsRequest request = fakeDataService.popLastRequest();
+    assertEquals(FULL_TABLE_NAME, request.getTableName());
+
+    AtomicLong readRowFailure =
+        fakeMetricRegistry.results.get("google-cloud-bigtable.grpc.method.ReadRow.failure");
+    AtomicLong operationLatency =
+        fakeMetricRegistry.results.get(
+            "google-cloud-bigtable.grpc.method.ReadRow.operation.latency");
+    AtomicLong tableGetLatency =
+        fakeMetricRegistry.results.get("google-cloud-bigtable.table.get.latency");
+
+    Assert.assertEquals(3, readRowFailure.get());
+    Assert.assertTrue(
+        "operation latency for ReadRow took longer than expected",
+        operationLatency.get() <= readRowsTime + 50);
+    Assert.assertTrue(
+        "operation latency for table.get took longer than expected",
+        tableGetLatency.get() <= readRowsTime + 50);
```

Comment:
this can be a lot tighter of a check. I would put a 5ms sleep in the fake service and bound this to 20ms

Reply:
Also, please add checks for the min bound for both measurements